slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -436,20 +515,33 @@ void SLAMServiceImpl::RunSLAM() {
     bool set_start_time = false;
     auto file = GetNextDataFile();
 
-    // define tmp_global_pose here so it always has the previous pose
+    // Define tmp_global_pose here so it always has the previous pose
     cartographer::transform::Rigid3d tmp_global_pose =
         cartographer::transform::Rigid3d();
     while (file != "") {
+        // Ignore files that are not *.pcd files
         if (file.find(".pcd") == std::string::npos) {
             file = GetNextDataFile();
             continue;
         }
+        // Go past files that are not supposed to be included in this run
+        if (!set_start_time &&
```

Comment:
Agree this would be easier to read, but leaving it in the if statement so the `ReadTimeFromFilename` function will be only executed if `set_start_time != true`

Reply:
Then consider combining this `!set_start_time check` with the one on line 531 and having an internal if statement for `ReadTimeFromFilename`. 

Will result in the same number of if statements but will keep them simpler.

Reply:
also will provide a clean read for seeing the updating of the `file` variable